### DISCUSSION TITLE
LLaMA Server combines the power of LLaMA C++ with the beauty of Chatbot UI

### DISCUSSION THREAD

**Discussion by nuance1979** [ggml-org/llama.cpp#733] [Show and tell]
Hi folks, I wrote [LLaMA Server](https://github.com/nuance1979/llama-server), a small http server to mimic the OpenAI APIs so that you can use [Chatbot UI](https://github.com/mckaywrigley/chatbot-ui), an open source implementation of ChatGPT UI, to interact with a LLaMA C++ instance. Here is a demo:

https://user-images.githubusercontent.com/10931178/229408428-5b6ef72d-28d0-427f-ae83-e23972e2dcff.mov

Hope you enjoy it! Cheers!

**Comment by gaemus** [comment]
In order to run the Llama version, you first need to input your OpenAI API key - then you can switch the Model to Llama.  This should be mentioned in the `README`. It would be better if you could access the Llama model without an OpenAI API key.
